src/main/java/com/amplitude/Retry.java

```diff
@@ -151,30 +147,28 @@ private static void retryEventsOnLoop(String userId, String deviceId, String api
                                 int numEventsRemoved = 0;
                                 for (int i = 0; i < eventIndicesToRemove.length; i++) {
                                     int index = eventIndicesToRemove[i];
-                                    if (index < eventCountHolder[0]) {
+                                    if (index < eventCount) {
                                         eventsBuffer.remove(i);
                                         numEventsRemoved += 1;
                                     }
                                 }
-                                eventCountHolder[0] -= numEventsRemoved;
-                                eventsInRetry -= eventCountHolder[0];
+                                eventCount -= numEventsRemoved;
+                                eventsInRetry -= eventCount;
                                 // If we managed to remove all the events, break early
-                                if (eventCountHolder[0] < 1) {
+                                if (eventCount< 1) {
                                     break;
                                 }
                             }
                             if (!shouldRetry) {
                                 break;
                             }
                             if (shouldReduceEventCount && !isLastTry) {
-                                eventCountHolder[0] = eventCountHolder[0] / 2;
+                                eventCount /= 2;
                             }
-                            // Clean up the events
-                            eventsBuffer.subList(0, eventCountHolder[0]).clear();
-                            eventsInRetry -= eventCountHolder[0];
                         } catch (InterruptedException e) {
                         }
                     }
+                    eventsInRetry -= eventCount;
```

Comment:
Since the logic in node js in here is cleaning up the eventsBuffer after the retry loop for all cases, so we don't need to put the eventsBuffer back to the idToBuffer map. 
For reference: https://github.com/amplitude/Amplitude-Node/blob/1f6d6578aa060ecd53f867721924b75b2ac22e1c/packages/node/src/retry/defaultRetry.ts#L272

Reply:
+1, Alyssa and I discussed this. It appears all the events are being removed anyway.